pkg/locker/locker.go

```diff
@@ -90,6 +93,13 @@ func (l *Locker) Lock(name string) {
 	// once locked then we can decrement the number of waiters for this lock
 	nameLock.Lock()
```

Comment:
Isn't this where it's going to block at?

I think we'll need to consider a different implementation, possibly a condition variable that can broadcast to all waiters on error or signal a single waiter when the lock is ready.